Store summaries in DB, rather than files

jeremyh opened the pull request:
More polish can still be done, but it now has feature-parity with the current version, the tests are _significantly_ better than before, and it's clearly the way forward now. I prefer not to keep branches open for so long that are this far-reaching across the codebase.

jeremyh committed:
Start of db summary store

jeremyh committed:
Add geometry conversion to db store

jeremyh committed:
Allow sharing of engine

jeremyh committed:
Enable Postgres service in Travis

jeremyh committed:
Add postgis package to Travis

travis-ci/travis-ci#6972

jeremyh committed:
Add test for empty period, to avoid previous bug

jeremyh committed:
Avoid loading products individually

The get_by_name() function is no longer cached in the new
version of datacube, so it's resulting in many small queries.

jeremyh committed:
Start experimental native dataset.extent implementation

jeremyh committed:
Add crs calculation

jeremyh committed:
Add server timings http headers during dev

jeremyh committed:
Add experimental decorator for index call timings

jeremyh committed:
Add proto dataset spatial table

jeremyh committed:
Fix return types of alchemy extent queries

jeremyh committed:
Update extents per-product, expand printed output

jeremyh committed:
Avoid double doc offset calculation

jeremyh committed:
Code readability improvement: spatial prototype.

jeremyh committed:
Clearer printing of spatial fields

jeremyh committed:
Store SRID rather than CRS string, add bounds column

jeremyh committed:
Cleanup printing, table population

jeremyh committed:
Move to single footprint column

jeremyh committed:
Add views for extent storage information

jeremyh committed:
Improve debug print formatting

jeremyh committed:
Add grid point calculation for datasets

jeremyh committed:
Remove stray print statements

jeremyh committed:
Disable default spatial index until tested

jeremyh committed:
Use postgres point type instead of Array(2)

jeremyh committed:
Add product extent materialized view

jeremyh committed:
Simplify usage of pg point

jeremyh committed:
Add grid extent to product_extent view

jeremyh committed:
Add store tests

jeremyh committed:
Add an index to SRS table

jeremyh committed:
Simplify generated grid cell query

It doesn't need the entire "bounds" polygon creation expression just for the center point.

jeremyh committed:
Use a composite type instead of built-in point

Point is missing a lot of basic stuff, like equality,
and it's significantly larger in size (16 bytes instead of 4)

jeremyh committed:
Only amend tables when schema doesn't exist

jeremyh committed:
Fix GridCell json conversion

jeremyh committed:
WIP merge extent gen into DB Store

jeremyh committed:
Add extent building to db store

jeremyh committed:
Union footprints in srid groups

jeremyh committed:
Remove file store

Adds complexity having multiple; and there'll be little reason to
use it once DB store is stable.

jeremyh committed:
Change default to DB store

jeremyh committed:
Add srid, timeline counts to db summary

jeremyh committed:
Fix many edge cases (db store). Tests almost pass.

jeremyh committed:
Maintain dataset polygons when grouping

Fixes test

jeremyh committed:
Treat timezones consistently with the old storage

Tests now pass. Grouping by the configured timezone probably
makes more sense than UTC, but it'll be revisited in a separate
commit.

jeremyh committed:
Improve verbosity of storage test output

jeremyh committed:
Calc all stats in the configured time zone

jeremyh committed:
Use higher-level api in integration test

To cover more.

jeremyh committed:
Centralise dodgy code. Formatting improvements.

jeremyh committed:
Avoid duplicate imports

jeremyh committed:
Check postgres type exists before creating

jeremyh committed:
Schema-prefix custom postgres type

jeremyh committed:
Add command to run/test extent code alone

jeremyh committed:
Update `generate` command for DB store. Add multiproc.

jeremyh committed:
Allow day summaries to be generated

jeremyh committed:
Fix invalid geojson for dataset list

jeremyh committed:
Close indexes in gen workers

jeremyh committed:
Don't do a full refresh on extent test tool

jeremyh committed:
Use center time everywhere stats, rather than start time

jeremyh committed:
Restore init()'s behaviour of initing all products

Fixes test failure.

jeremyh committed:
Show grid cell for each hovered dataset

jeremyh committed:
Fix test for dataset FeatureCollection

jeremyh committed:
Flatten the Store hierarchy

Only one implementation now: prefer simplicity.

jeremyh committed:
Avoid duplicating where-clause

jeremyh committed:
Add backend for per-grid-cell counts

jeremyh committed:
Fix GriCell Array conversion

jeremyh committed:
Fix incorrect index definition

jeremyh committed:
Fix alchemy type error message

jeremyh committed:
Fix gridcell loading from postgres

jeremyh committed:
Fix loading of gridcell arrays

Using lower-level psycopg2 seems much simpler.

jeremyh committed:
Remove dataset geojson from summary storage

It takes up a lot of space, and is largely only used by the on-demand generated ones.

jeremyh committed:
Fix bad indentation

jeremyh committed:
Add "show" cli command for testing summaries

jeremyh committed:
Tighten nullability of spatial table

Also reshuffle to put larger and nullable columns last,
for imaginary performance gain.

jeremyh committed:
Simplify empty product handling

jeremyh committed:
Fix empty aggregates (failed tests)

jeremyh committed:
Fix empty-period test

jeremyh committed:
Record the time of last product refresh

jeremyh committed:
Remove unused code

jeremyh committed:
Clear product cache after update

Fixes store test failures

jeremyh committed:
Remove print logging

jeremyh committed:
Track the size of datasets (tests fail)

jeremyh committed:
Show file size of selections

jeremyh committed:
Update expected calculated values of storage tests

jeremyh committed:
Split summary storage logic from calculation

jeremyh committed:
Expand tests for empty products and periods

jeremyh committed:
Fix updates of summary gen time

jeremyh committed:
Expand tests to cover the generate command

jeremyh committed:
Fix test configuration of 'generate' command

jeremyh committed:
Avoid caching missing-products

Fixes tests.

Missing-product is a rare occurrance, so not worth caching, and is tripping up tests that check for a product existing before adding them from a separate process.

jeremyh committed:
Return an error code for unknown products

jeremyh committed:
Timezone consistency for recorded time ranges

jeremyh committed:
Fix store tests: inconsistent timezone handling

jeremyh committed:
Gridcell has incorrect type hint

jeremyh committed:
Fix lack of gridcell type conversion on load

When not init()'ing, the gridcell type wasn't registered with psycopg2.

jeremyh committed:
Fix missing gen_time on day stats

jeremyh committed:
Start of grid count ui

jeremyh committed:
Separate the init/create/refresh concepts in Store

jeremyh committed:
Fix broken constructor usage

jeremyh committed:
Use higher-level api when possible

jeremyh committed:
Use engines consistently

jeremyh committed:
Avoid making any connections on store construction

jeremyh committed:
Update day-summary test: days aren't stored

jeremyh committed:
Simplify unioned footprints to a tolerance

Indistinguishable on web map, but significantly smaller.

jeremyh committed:
Constrain grid counts by the footprint

jeremyh committed:
Avoid having multiple references to the index

jeremyh committed:
Fix periods with no footprint

jeremyh committed:
Handle failing extent calculations

eg. ls5_nbar_albers 1999

jeremyh committed:
Speed up grid polygon generation

jeremyh committed:
Improve convenience of get/has/update api

Most fields optional

jeremyh committed:
Fix no-result and 404 pages

jeremyh committed:
Add page-load tests for wofs datasets

jeremyh committed:
Add wofs sample data for integration tests

jeremyh committed:
Use utc for canonical date

jeremyh committed:
Limit decimal points of displayed float values

jeremyh committed:
Fix index override in integration tests

jeremyh committed:
Check product and about pages for load errors

jeremyh committed:
Remove unused arg from generate command

jeremyh committed:
Rename gridcell to region

Based on discussion with Kirill

jeremyh committed:
Remove old gridcell type tests

jeremyh committed:
Update tests for grid->region rename

jeremyh committed:
Add path/row region geometry

jeremyh committed:
Use color instead of opacity. Needs polish.

jeremyh committed:
More useful logging for unsupported regions

jeremyh committed:
Avoid regions on products with invalid grid_spec

Such as high_tide_comp_20p

jeremyh committed:
Fix timeline control after model renames

jeremyh committed:
Disable region clickability until implemented

jeremyh committed:
Show region count

jeremyh committed:
Improve grammar "None visible"

jeremyh committed:
Add check that area is nonzero

jeremyh committed:
experimental: store footprints as albers

jeremyh committed:
Remove debug loggin

jeremyh committed:
pep8 fix

jeremyh committed:
Fix tests for empty overviews

jeremyh committed:
Add srid property test

jeremyh committed:
Improve empty footprint test error message

jeremyh committed:
Test: Unconstrain the test geometry

jeremyh committed:
Set column rather than value srid

jeremyh committed:
Validate that srid is set

jeremyh committed:
Fix missing srid

geoalchemy seems to use column definition srid: ignores value srid?

jeremyh committed:
Fix overview rendering for empty periods

jeremyh committed:
Reorganise total tally fields

jeremyh committed:
Adjust simplify tolerance for albers footprints

Different units: meters

jeremyh committed:
Always include empty days in timeline

jeremyh committed:
Fix page load on empty region counts

jeremyh committed:
Check for empty region counts

jeremyh committed:
Fix handling of datasets without CRSes (eg. hltc)

jeremyh committed:
Consistent indent in extent calc

jeremyh committed:
Fix test case: empty region count, full time count

jeremyh committed:
Expand leeway for footprint area test

A 0.1% tolerance. We're not rounding, and it's right on the edge of 8/9.

jeremyh committed:
Test failure: region_dataset_count loading

jeremyh committed:
Merge remote-tracking branch 'upstream/develop' into jez/native_spatial

jeremyh committed:
Update reports for the recent property renames

jeremyh committed:
Move region logic out of the view

jeremyh committed:
Re-add label to individual dataset geojson

jeremyh committed:
Update API for three polygon types

jeremyh committed:
WIP, start of polygon type selection

ashoka1234 commented:
Summaries in DB probably makes sense and I think it makes further feature development bit easier and storage efficient.  When summaries get larger, database queries could be faster or made faster relative to file store, I guess.

jeremyh committed:
Clean-up UI for choosing type of view

jeremyh committed:
Fix tests for empty summary regions

jeremyh committed:
Remove import duplication

jeremyh committed:
Fix no-data region test

jeremyh committed:
Fix empty regions of timeline

jeremyh committed:
Fix "zero data" test case for whole products

jeremyh committed:
Fix timeline test: some empty records not stored

jeremyh committed:
Remove recently-added check

Flagrantly disable a low-value check that's not worth the effort at the moment.

jeremyh committed:
Fix low tide composite dataset display

jeremyh committed:
Clean up duplicate imports

jeremyh committed:
Consistent font across platforms

jeremyh committed:
WIP clearer region display

jeremyh committed:
Add "RegionInfo" as a concept

jeremyh committed:
Enable doctests in pytest

jeremyh committed:
Improve display of undisplayable products

Such as telemetry data products: No CRS, no footprints.

jeremyh committed:
Tweak chart saturation

Match the grids, which are less saturated than previously (they must be transparent)

jeremyh committed:
Pluralise 'datasets' correctly

jeremyh committed:
Remove duplicate click handler

jeremyh committed:
Remove incorrect hover 'sample' message

jeremyh committed:
Make page load tests indifferent to whitespace

jeremyh committed:
Add tests for file size calc, undisplayable products

jeremyh committed:
Allow for environment include. Update README.

jeremyh committed:
Update dependencies: requires ODC 1.6

And the ship has sailed on Python 3.6. Sorry!

jeremyh committed:
Add README directions for running integration tests

jeremyh committed:
Add screenshot to readme
